src/trans.rs

```diff
@@ -719,6 +727,48 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                 }
             }
 
+            Rvalue::CheckedBinaryOp(ref op, ref left, ref right) => {
+                // TODO: This shouldn't just be a copy BinaryOp above!
+                // We should do some actual _checking_!
```

Comment:
Having that emulation written in Rust instead of generating it? Depending on what the internal API is of mir2wasm, it could be just as easy to emit the right code here (and you wouldn't have compiler-implementation dependencies).